### PR TITLE
EDU-1023: tag Go testing nodes

### PR DESCRIPTION
**Pull request by DJSanti**
## What does this PR do?
This PR adds tags to the source nodes for the Testing section of the Go SDK.

### CODE REVIEW

File: docs-src/go/listen-to-heartbeats.md [hunk 1]
```diff
@@ -4,7 +4,12 @@ title: Listen to Heartbeats
 description: When an Activity sends a Heartbeat, be sure that you can see the Heartbeats in your test code so that you can verify them.
 sidebar_label: Listen to Heartbeats
 tags:
-  - guide-context
+  - developer-guide-doc-type
+  - go sdk
+  - testing
+  - testing-guide-doc-type
```
Comment (flossypurse) [inline docs-src/go/listen-to-heartbeats.md:10]: i like `how-to doc type` and `explanation doc-type` and `reference doc-type`

Not sure about these others at this point.

File: docs-src/go/mock-activities.md [hunk 2]
```diff
@@ -4,7 +4,12 @@ title: How to mock Activities
 description: Mock an Activity by providing mock Activity implementations to the Worker.
 sidebar_label: Mock Activities
 tags:
-  - guide-context
+  - developer-guide-doc-type
+  - go sdk
+  - testing
+  - testing-guide-doc-type
+  - activities
```
Comment (flossypurse) [inline docs-src/go/mock-activities.md:11]:
```suggestion
  - activity
```
I think we decided to use singular in these cases.

File: docs-src/go/skip-time-set-up.md [hunk 3]
```diff
@@ -4,7 +4,11 @@ title: Set up time skipping
 description: The test server included in most SDKs is an in-memory implementation of Temporal Server that supports skipping time.
 sidebar_label: Setting up
 tags:
-  - guide-context
+  - developer-guide-doc-type
+  - go sdk
+  - testing
+  - testing-guide-doc-type
+  - workflows
```
Comment (flossypurse) [inline docs-src/go/skip-time-set-up.md:11]: singular

**Review by flossypurse — CHANGES_REQUESTED** [review]

**Review by flossypurse — CHANGES_REQUESTED** [review]
Just curious what is the difference in your opinion between these tags?
  - testing
  - testing-guide-doc-type

**Review by napcs — APPROVED** [review]

**Pull request merged**